presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSessionProperties.java

```diff
@@ -76,6 +76,7 @@
     private static final String MAX_SPLIT_SIZE = "max_split_size";
     private static final String MAX_INITIAL_SPLIT_SIZE = "max_initial_split_size";
     private static final String RCFILE_OPTIMIZED_WRITER_VALIDATE = "rcfile_optimized_writer_validate";
+    private static final String RCFILE_PAGESOURCE_VALIDATE_FILE_SIZE = "rcfile_page_source_validate_file_size";
```

Comment:
Page source is an implementation detail that we would avoid exposing to end users. How about something like `rcfile_validate_file_size_on_read`